Source/Core/InputCommon/ControllerInterface/Touch/Touchscreen.cpp

```diff
@@ -0,0 +1,240 @@
+  AddInput(new Button(_padID, ButtonManager::GUITAR_STRUM_DOWN));
+  AddInput(new Axis(_padID, ButtonManager::GUITAR_STICK_LEFT));
+  AddInput(new Axis(_padID, ButtonManager::GUITAR_STICK_RIGHT));
+  AddInput(new Axis(_padID, ButtonManager::GUITAR_STICK_UP));
+  AddInput(new Axis(_padID, ButtonManager::GUITAR_STICK_DOWN));
+  AddInput(new Axis(_padID, ButtonManager::GUITAR_WHAMMY_BAR));
+
+  // Wii-ext: Drums
+  AddInput(new Button(_padID, ButtonManager::DRUMS_BUTTON_MINUS));
+  AddInput(new Button(_padID, ButtonManager::DRUMS_BUTTON_PLUS));
+  AddInput(new Button(_padID, ButtonManager::DRUMS_PAD_RED));
+  AddInput(new Button(_padID, ButtonManager::DRUMS_PAD_YELLOW));
+  AddInput(new Button(_padID, ButtonManager::DRUMS_PAD_BLUE));
+  AddInput(new Button(_padID, ButtonManager::DRUMS_PAD_GREEN));
+  AddInput(new Button(_padID, ButtonManager::DRUMS_PAD_ORANGE));
+  AddInput(new Button(_padID, ButtonManager::DRUMS_PAD_BASS));
+  AddInput(new Axis(_padID, ButtonManager::DRUMS_STICK_LEFT));
+  AddInput(new Axis(_padID, ButtonManager::DRUMS_STICK_RIGHT));
+  AddInput(new Axis(_padID, ButtonManager::DRUMS_STICK_UP));
+  AddInput(new Axis(_padID, ButtonManager::DRUMS_STICK_DOWN));
+
+  // Wii-ext: Turntable
+  AddInput(new Button(_padID, ButtonManager::TURNTABLE_BUTTON_GREEN_LEFT));
+  AddInput(new Button(_padID, ButtonManager::TURNTABLE_BUTTON_RED_LEFT));
+  AddInput(new Button(_padID, ButtonManager::TURNTABLE_BUTTON_BLUE_LEFT));
+  AddInput(new Button(_padID, ButtonManager::TURNTABLE_BUTTON_GREEN_RIGHT));
+  AddInput(new Button(_padID, ButtonManager::TURNTABLE_BUTTON_RED_RIGHT));
+  AddInput(new Button(_padID, ButtonManager::TURNTABLE_BUTTON_BLUE_RIGHT));
+  AddInput(new Button(_padID, ButtonManager::TURNTABLE_BUTTON_MINUS));
+  AddInput(new Button(_padID, ButtonManager::TURNTABLE_BUTTON_PLUS));
+  AddInput(new Button(_padID, ButtonManager::TURNTABLE_BUTTON_HOME));
+  AddInput(new Button(_padID, ButtonManager::TURNTABLE_BUTTON_EUPHORIA));
+  AddInput(new Axis(_padID, ButtonManager::TURNTABLE_TABLE_LEFT_LEFT));
+  AddInput(new Axis(_padID, ButtonManager::TURNTABLE_TABLE_LEFT_RIGHT));
+  AddInput(new Axis(_padID, ButtonManager::TURNTABLE_TABLE_RIGHT_LEFT));
+  AddInput(new Axis(_padID, ButtonManager::TURNTABLE_TABLE_RIGHT_RIGHT));
+  AddInput(new Axis(_padID, ButtonManager::TURNTABLE_STICK_LEFT));
+  AddInput(new Axis(_padID, ButtonManager::TURNTABLE_STICK_RIGHT));
+  AddInput(new Axis(_padID, ButtonManager::TURNTABLE_STICK_UP));
+  AddInput(new Axis(_padID, ButtonManager::TURNTABLE_STICK_DOWN));
+  AddInput(new Axis(_padID, ButtonManager::TURNTABLE_CROSSFADE_LEFT));
+  AddInput(new Axis(_padID, ButtonManager::TURNTABLE_CROSSFADE_RIGHT));
+  AddInput(new Axis(_padID, ButtonManager::TURNTABLE_EFFECT_DIAL));
+
+  // Rumble
+  AddOutput(new Motor(_padID, ButtonManager::RUMBLE));
+}
+// Buttons and stuff
+
+std::string Touchscreen::Button::GetName() const
+{
+  std::ostringstream ss;
+  ss << "Button " << (int)_index;
```

Comment:
This should use a `static_cast` instead of a c-style one, since it is a simple conversion that would otherwise (potentially) attempt various types of conversions.
```suggestion
  ss << "Button " << static_cast<int>(_index);
```

Same further down for `Axis` and `Rumble`